mongodbatlas/clusters.go

```diff
@@ -136,32 +136,36 @@ type ConnectionStrings struct {
 
 // Cluster represents MongoDB cluster.
 type Cluster struct {
-	AutoScaling              *AutoScaling             `json:"autoScaling,omitempty"`
-	BackupEnabled            *bool                    `json:"backupEnabled,omitempty"` // Deprecated: Use ProviderBackupEnabled instead
-	BiConnector              *BiConnector             `json:"biConnector,omitempty"`
-	ClusterType              string                   `json:"clusterType,omitempty"`
-	DiskSizeGB               *float64                 `json:"diskSizeGB,omitempty"`
-	EncryptionAtRestProvider string                   `json:"encryptionAtRestProvider,omitempty"`
-	Labels                   []Label                  `json:"labels,omitempty"`
-	ID                       string                   `json:"id,omitempty"`
-	GroupID                  string                   `json:"groupId,omitempty"`
-	MongoDBVersion           string                   `json:"mongoDBVersion,omitempty"`
-	MongoDBMajorVersion      string                   `json:"mongoDBMajorVersion,omitempty"`
-	MongoURI                 string                   `json:"mongoURI,omitempty"`
-	MongoURIUpdated          string                   `json:"mongoURIUpdated,omitempty"`
-	MongoURIWithOptions      string                   `json:"mongoURIWithOptions,omitempty"`
-	Name                     string                   `json:"name,omitempty"`
-	NumShards                *int64                   `json:"numShards,omitempty"`
-	Paused                   *bool                    `json:"paused,omitempty"`
-	PitEnabled               *bool                    `json:"pitEnabled,omitempty"`
-	ProviderBackupEnabled    *bool                    `json:"providerBackupEnabled,omitempty"`
-	ProviderSettings         *ProviderSettings        `json:"providerSettings,omitempty"`
-	ReplicationFactor        *int64                   `json:"replicationFactor,omitempty"`
-	ReplicationSpec          map[string]RegionsConfig `json:"replicationSpec,omitempty"`
-	ReplicationSpecs         []ReplicationSpec        `json:"replicationSpecs,omitempty"`
-	SrvAddress               string                   `json:"srvAddress,omitempty"`
-	StateName                string                   `json:"stateName,omitempty"`
-	ConnectionStrings        *ConnectionStrings       `json:"connectionStrings,omitempty"`
+	*BaseCluster
```

Comment:
[q] why pointer here and no pointer on the advanced one?

Reply:
before merging this, can you try and update mongocli to this branch and see if it builds/compile? I know we'll need changes based on the url change from your previous PR but we should try to make this a backward compatible change so that we don't need to re implement how we create clusters (and how others may be using this lib)

Reply:
Done! : )